pkgs/development/python-modules/laszip/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, scikit-build-core
+, distlib
+, pytestCheckHook
+, pyproject-metadata
+, pathspec
+, pybind11
+, cmake
+, LASzip
+}:
+
+buildPythonPackage rec {
+  pname = "laszip-python";
+  version = "0.2.1";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    rev = version;
+    repo = pname;
+    owner = "tmontaigu";
+    sha256 = "sha256-ujKoUm2Btu25T7ZrSGqjRc3NR1qqsQU8OwHQDSx8grY=";
+  };
+
+  propagatedBuildInputs = [
+    scikit-build-core
+    distlib
+    pyproject-metadata
+    pathspec
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    pybind11
+    LASzip
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  preBuild = ''
+    cd ..
+  '';
+
+  doCheck = false;
```

Comment:
whats wrong with the tests? Im just curious

Reply:
there are no tests, which causes pytest to fail, which causes the stdenv setup script to fail in a strange way:

```
python3.10-laszip-python> pytestCheckPhase
python3.10-laszip-python> Executing pytestCheckPhase
python3.10-laszip-python> ============================= test session starts ==============================
python3.10-laszip-python> platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
python3.10-laszip-python> rootdir: /build/source
python3.10-laszip-python> collected 0 items                                                              
python3.10-laszip-python> ============================ no tests ran in 0.00s =============================
python3.10-laszip-python> /nix/store/chilfhdcsnmwjl7igrw26j1lrc0zar35-stdenv-linux/setup: line 1570: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/y5yxbycasvpv9idk4c00vd8alzmsnpa0-python3.10-laszip-python-0.2.1.drv' failed with exit code 5;
       last 10 log lines:
       > Check whether the following modules can be imported: laszip
       > pytestCheckPhase
       > Executing pytestCheckPhase
       > ============================= test session starts ==============================
       > platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
       > rootdir: /build/source
       > collected 0 items                                                              
       >
       > ============================ no tests ran in 0.00s =============================
       > /nix/store/chilfhdcsnmwjl7igrw26j1lrc0zar35-stdenv-linux/setup: line 1570: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/y5yxbycasvpv9idk4c00vd8alzmsnpa0-python3.10-laszip-python-0.2.1.drv'.
```

Reply:
Can you add a comment that there are no tests in that case.

Note the test runner sometimes simply can't find them, doesn't mean they don't exist.